adapter.go

```diff
@@ -165,6 +165,10 @@ func NewAdapterByDBUseTableName(db *gorm.DB, prefix string,tablename string) (*A
 }
 
 func NewAdapterByDB(db *gorm.DB) (*Adapter, error) {
+	if db != nil && db.Statement.Table == "" {
+		db = db.Scopes(CasbinTableName(&CasbinRule{TableName: "casbin_rule"}))
```

Comment:
you must add `Session(&gorm.Session{WithConditions: true})` after the `db.Scopes()`

Reply:
refer to L250-L251